File: cross_lang/memory/linked_list.c
#include <stdio.h>
#include <stdlib.h>

typedef struct ListNode {
    long long val;
    struct ListNode *next;
} ListNode;

int main() {
    long long count = 50000;
    ListNode *head = NULL;
    for (long long i = 0; i < count; i++) {
        ListNode *node = (ListNode *)malloc(sizeof(ListNode));
        node->val = i;
        node->next = head;
        head = node;
    }

    long long sum = 0;
    ListNode *cur = head;
    while (cur != NULL) {
        sum += cur->val;
        ListNode *tmp = cur;
        cur = cur->next;
        free(tmp);
    }
    printf("%lld\n", sum);
    return 0;
}

File: cross_lang/memory/linked_list.js
class ListNode {
    constructor(val, next) {
        this.val = val;
        this.next = next;
    }
}

const count = 50000;
let head = null;
for (let i = 0; i < count; i++) {
    head = new ListNode(i, head);
}

let sum = 0;
let cur = head;
while (cur !== null) {
    sum += cur.val;
    cur = cur.next;
}
console.log(sum);
